src/components/big-interactive-pages/editor.tsx

```diff
@@ -66,9 +68,23 @@ let lastSavePromise = Promise.resolve();
 let saveQueueSize = 0;
 export const saveGame = debounce(
 	800,
-	(persistenceState: Signal<PersistenceState>, code: string) => {
+	(persistenceState: Signal<PersistenceState>, code: string, sessionId: string) => {
 		const doSave = async () => {
 			const attemptSaveGame = async () => {
+				const game = (persistenceState.value.kind === 'PERSISTED' && persistenceState.value.game !== 'LOADING') ? persistenceState.value.game : null;
+				if (!game) return false;
+				const lastSavedSessionInfo = localStorage.getItem('lastSavedSessionId');
+				const lastSavedData = lastSavedSessionInfo ? JSON.parse(lastSavedSessionInfo) : null;
+				const lastSavedSessionId = lastSavedData ? lastSavedData.sessionId : null;
+				const lastSavedGameId = lastSavedData ? lastSavedData.gameId : null;
+				
+				console.log('Attempting to save. Current session:', sessionId, 'Last saved session:', lastSavedSessionId);
+				
+				if (lastSavedGameId === game.id && lastSavedSessionId !== sessionId) {
+					showSaveConflictModal.value = true;
+					continueSaving.value = false;
+					return false;
```

Comment:
TBH, I think a better, high-level pattern here would be the following:

```
if(shouldShowConflict(cxt)) {
  showConflict(cxt);
} else {
  if (!save(cxt)) {
    // show error
  }
}
```

Instead, what we're working with here is:
```
if (!save(cxt)) {
  // Figure out why saving failed, it left us a breadcrumb multivariable trail of clues
  if (cxt.continueSaving) { // logic } else if (cxt.showSaveConflictModal) { //logic }....
```